src/audio/codec_adapter/codec/cadence.c

```diff
@@ -122,22 +122,22 @@ int cadence_codec_init(struct comp_dev *dev)
 	if (ret != LIB_NO_ERROR) {
 		comp_err(dev, "cadence_codec_init() error %x: failed to get lib name",
 			 ret);
-		codec_free_memory(dev, cd);
+		pipeline_devm_free(dev->pipline);
```

Comment:
can we just confirm that this is intended and correct - before the change this was freeing one specific object, now after this change this is freeing all allocated blocks?

Reply:
All pipeline objects will be in a block, but this does need checking here.